pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -629,13 +651,72 @@ func (b *OFBridge) RetryInterval() time.Duration {
 	return b.retryInterval
 }
 
+func (b *OFBridge) queryTableFeatures() {
+	mpartRequest := &openflow13.MultipartRequest{
+		Header: openflow13.NewOfp13Header(),
+		Type:   openflow13.MultipartType_TableFeatures,
+		Flags:  0,
+	}
+	mpartRequest.Header.Type = openflow13.Type_MultiPartRequest
+	mpartRequest.Header.Length = mpartRequest.Len()
+	tableFeatureChan := make(chan *openflow13.MultipartReply)
+	b.mpartRplChs[mpartRequest.Xid] = tableFeatureChan
+	go func() {
+		// Delete the channel which is used to receive the MultipartReply message after all table features are received.
+		defer func() {
+			delete(b.mpartRplChs, mpartRequest.Xid)
+			close(tableFeatureChan)
+		}()
+		b.processTableFeatures(tableFeatureChan)
+	}()
+	b.ofSwitch.Send(mpartRequest)
+}
+
+func (b *OFBridge) processTableFeatures(ch chan *openflow13.MultipartReply) {
+	header := openflow13.NewOfp13Header()
+	header.Type = openflow13.Type_MultiPartRequest
+	complete := false
+	// Since the initial MultipartRequest doesn't specify any table ID, OVS will reply all tables' (except the hidden one)
+	// features in the reply. Here we complete the loop after we receive all the reply messages, while the reply message
+	// is configured with Flags=0.
+	for !complete {
+		select {
+		case rpl := <-ch:
+			request := &openflow13.MultipartRequest{
+				Header: header,
+				Type:   openflow13.MultipartType_TableFeatures,
+				Flags:  rpl.Flags,
+			}
+			// A MultipartReply message may have one or many OFPTableFeatures messages, and MultipartReply.Body is a
+			// slice of these messages.
+			for _, body := range rpl.Body {
+				tableFeature := body.(*openflow13.OFPTableFeatures)
+				// Modify table name if the table is in the pipeline, otherwise use the default table features.
+				// OVS doesn't allow to skip any table except the hidden table (always the last table) in a table_features
+				// request. So use the existing table features for the tables that Antrea doesn't define in the pipeline.
+				if t, ok := b.tableCache[tableFeature.TableID]; ok {
+					// Set table name with the configured value.
+					copy(tableFeature.Name[0:], t.name)
+				}
+				request.Body = append(request.Body, tableFeature)
+			}
+			request.Length = request.Len()
+			b.ofSwitch.Send(request)
+			if rpl.Flags == 0 {
```

Comment:
add a comment to indicate that this is guaranteed to be the last message in the multipart reply, as per the OF spec. I know you have one at the beginning of the loop, but it belongs here IMO.